delocate/delocating.py

```diff
@@ -653,6 +654,14 @@ def _get_archs_and_version_from_wheel_name(
             raise ValueError(f"Invalid platform tag: {platform_tag.platform}")
         major, minor, arch = match.groups()
         platform_requirements[arch] = Version(f"{major}.{minor}")
+    # If we have a wheel name with arm64 and x86_64 we have to convert that to
+    # universal2
+    if platform_requirements.keys() == {"arm64", "x86_64"}:
+        version = platform_requirements["arm64"]
+        if version == Version("11.0"):
+            version = platform_requirements["x86_64"]
+        platform_requirements = {"universal2": version}
```

Comment:
This branch is not covered by tests at the moment. I assumed that delocating a universal wheel would meet this condition but it seems I was wrong.

Reply:
I've added a test for this case in the `test_get_archs_and_version_from_wheel_name` test. Delocating a universal2 wheel doesn't enter this path because the tag from the name is already universal2, not one tag with arm64 and another with x86_64, which is what this fixes.